test/distributed/test_c10d_spawn.py

```diff
@@ -241,11 +241,7 @@ def _test_base(self, net, inp, check_allclose=True):
         store = c10d.FileStore(self.file.name, self.world_size)
         process_group = c10d.ProcessGroupGloo(store, self.rank, self.world_size)
         if inp[0].is_cuda:
-            num_gpus = torch.cuda.device_count()
-            batch_size = inp[0].size(0)
-            # batch_size must be evenly divisible by num_gpus_used, take the largest one
-            num_gpus_used = [i for i in range(1, num_gpus + 1) if batch_size % i == 0][-1]
-            device_ids = list(range(num_gpus_used))
+            device_ids = [torch.cuda.current_device()]
```

Comment:
I'm wondering whether we can check this test throws expected errors when using SPMD? after SPMD is fully killed, we can remove this test?

Reply:
Added a few tests in test_c10d.py.